README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+
+```go
+user := FindLastPoster()
+rowsUpdated, err := store.Update(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Save models
+
+To save a model we just need to use the `Save` method of the store and pass it a model. It will update the model if it was already persisted or insert it otherwise.
+
+```go
+updated, err := store.Save(user)
+if err != nil {
+        // handle error
+}
+
+if updated {
+        // it was updated, not inserted
+}
+```
+
+If our model has relationships, they will be saved as well. The relationships of the relationships will not, though. Relationships are only saved with one level of depth.
+
+```go
+user := NewUser("foo")
+user.Posts = append(user.Posts, NewPost(user, "new post"))
+
+updated, err := store.Save(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Delete models
+
+To delete a model we just have to use the `Delete` method of the store. It will return an error if the model was not already persisted.
+
+```go
+err := store.Delete(user)
+if err != nil {
+        // handle error
+}
+```
+
+Relationships of the model are **not** automatically removed using `Delete`.
+
+For that, specific methods are generated in the store of the model.
+
+For one to many relationships:
+
+```go
+// remove specific posts
+err := store.RemovePosts(user, post1, post2, post3)
+if err != nil {
+        // handle error
+}
+
+// remove all posts
+err := store.RemovePosts(user)
+```
+
+For one to one relationships:
+
+```go
+// remove the thing
+err := store.RemoveThing(user)
+```
+
+## Query models
+
+### Simple queries
+
+To perform a query you have to do the following things: 
+* Create a query
+* Pass the query to `Find`, `FindOne`, `MustFind` or `MustFindOne` of the store
+* Gather the results from the result set
+
+```go
+// Create the query
+q := NewUserQuery().
+        Where(kallax.Like(Schema.User.Username, "joe%")).
+        Order(kallax.Asc(Schema.User.Username)).
+        Limit(20).
+        Offset(2)
+
+rs, err := store.Find(q)
+if err != nil {
+        // handle error
+}
+
+for rs.Next() {
+        user, err := rs.Get()
+        if err != nil {
+                // handle error
+        }
+}
+```
+
+Next will automatically close the result set when it hits the end. If you have to prematurely exit the iteration you can close it manually with `rs.Close()`.
+
+You can query just a single row with `FindOne`.
+
+```go
+q := NewUserQuery().
+        Where(kallax.Eq(Schema.User.Username, "Joe"))
+
+user, err := store.FindOne(q)
+```
+
+By default, all columns in a row are retrieved. To not retrieve all of them, you can specify the columns to include/exclude.
```

Comment:
I will explain that retrieving partial objects will make them not writable, and then I'd add a link to the `reload` section